### PR TITLE
Exclude regions in export.

### LINKED ISSUE #1054: Export config
Add more options

* include all regions, but exclude a specific list
* only include non-travellers.

The existing setup works well for

* home country + travellers export

We need the ability to create a second export that represents

* foreign country non travelers

/kind enhancement

### PR DESCRIPTION
**Pull request by jeremyfaller**
Recreate the ability to exclude regions from the exports.

Updates #1054

## Proposed Changes

* Recreate ExcludeRegions in exports. This will allow situations where traveling users can be excluded in an export.

**Release Note**

```release-note
Recreate ExcludeRegions in exports. This will allow situations where traveling users can be excluded in an export.

```

### CODE REVIEW

**Comment by jeremyfaller** [comment]
I think this now actually ready for review.

/unhold

File: internal/admin/exports/form.go [hunk 1]
```diff
@@ -60,6 +61,13 @@ func (f *formData) PopulateExportConfig(ec *model.ExportConfig) error {
 		}
 	}
 	ec.IncludeTravelers = f.IncludeTravelers
+	ec.ExcludeRegions = make([]string, 0)
+	for _, s := range strings.Split(f.ExcludeRegions, "\n") {
```
Comment (mikehelmick) [inline internal/admin/exports/form.go:65]: can we extract this loop out to a function? duplicate code.

File: internal/export/database/export_test.go [hunk 2]
```diff
@@ -541,6 +542,109 @@ func TestTravelerKeys(t *testing.T) {
...
+		OutputRegion:   ec.OutputRegion,
+		Status:         model.ExportBatchOpen,
+		ExcludeRegions: ec.ExcludeRegions,
+	}
+	if err := New(testDB).AddExportBatches(ctx, []*model.ExportBatch{eb}); err != nil {
+		t.Fatal(err)
+	}
+
+	// Create person in main region.
+	trav := &publishmodel.Exposure{
```
Comment (mikehelmick) [inline internal/export/database/export_test.go:585]: this says `trav` but the exposure isn't marked as a traveler

**Review by mikehelmick — COMMENTED** [review]
This is only part of what #1054 needs - fine for the rest to be a followup.

The missing part is the "only include non-travellers" option

There is the "include travelers" option today, which lets one set up
 -home region + travelers
but we can't set up yet
- foreign regions, non-travelers

File: internal/admin/exports/form.go (hunk 1 above)
Comment (jeremyfaller) [reply internal/admin/exports/form.go]: done, and added test.

File: internal/export/database/export_test.go (hunk 2 above)
Comment (jeremyfaller) [reply internal/export/database/export_test.go]: my mistake

File: internal/admin/exports/form.go [hunk 3]
```diff
@@ -38,6 +39,19 @@ type formData struct {
 	SigInfoIDs       []int64       `form:"siginfo"`
 }
 
+// splitRegions turns a string of regions (generally separated by newlines), and
+// breaks them up into a slice of strings.
+func splitRegions(regions string) []string {
+	ret := make([]string, 0)
```
Comment (sethvargo) [inline internal/admin/exports/form.go:45]: This should probably have some sane starting capacity.

Comment (sethvargo) [reply internal/admin/exports/form.go:45]: And should we do any kind of sorting for consistency?

File: internal/admin/exports/form_test.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+)
+
+func TestSplitRegions(t *testing.T) {
+	tests := []struct {
+		r string
+		e []string
+	}{
+		{"", []string{}},
+		{"  test  ", []string{"test"}},
+		{"test\n\rfoo", []string{"test", "foo"}},
```
Comment (sethvargo) [inline internal/admin/exports/form_test.go:15]: Can we add a test for _just_ `\n` separated?

**Review by sethvargo — COMMENTED** [review]
Two Qs

File: internal/admin/exports/form_test.go (hunk 4 above)
Comment (jeremyfaller) [reply internal/admin/exports/form_test.go:15]: Done.

File: internal/admin/exports/form.go (hunk 3 above)
Comment (jeremyfaller) [reply internal/admin/exports/form.go:45]: Sure, NBD.

**Comment by mikehelmick** [comment]
/lgtm

Followup still needed: configuration for foreign regions, non-travelers

**Pull request merged**